File: household-ledger-mobile/lib/domain/entities/receipt.dart
enum OcrStatus {
  pending,
  processing,
  completed,
  failed,
}

class Receipt {
  final int id;
  final int? transactionId;
  final int userId;
  final int? groupId;
  final String imageUrl;
  final Map<String, dynamic>? ocrResult;
  final OcrStatus ocrStatus;
  final int? extractedAmount;
  final DateTime? extractedDate;
  final String? extractedMerchant;
  final List<Map<String, dynamic>>? extractedItems;
  final double? confidenceScore;
  final bool verified;
  final DateTime createdAt;
  final DateTime updatedAt;

  Receipt({
    required this.id,
    this.transactionId,
    required this.userId,
    this.groupId,
    required this.imageUrl,
    this.ocrResult,
    required this.ocrStatus,
    this.extractedAmount,
    this.extractedDate,
    this.extractedMerchant,
    this.extractedItems,
    this.confidenceScore,
    required this.verified,
    required this.createdAt,
    required this.updatedAt,
  });

  Receipt copyWith({
    int? id,
    int? transactionId,
    int? userId,
    int? groupId,
    String? imageUrl,
    Map<String, dynamic>? ocrResult,
    OcrStatus? ocrStatus,
    int? extractedAmount,
    DateTime? extractedDate,
    String? extractedMerchant,
    List<Map<String, dynamic>>? extractedItems,
    double? confidenceScore,
    bool? verified,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Receipt(
      id: id ?? this.id,
      transactionId: transactionId ?? this.transactionId,
      userId: userId ?? this.userId,
      groupId: groupId ?? this.groupId,
      imageUrl: imageUrl ?? this.imageUrl,
      ocrResult: ocrResult ?? this.ocrResult,
      ocrStatus: ocrStatus ?? this.ocrStatus,
      extractedAmount: extractedAmount ?? this.extractedAmount,
      extractedDate: extractedDate ?? this.extractedDate,
      extractedMerchant: extractedMerchant ?? this.extractedMerchant,
      extractedItems: extractedItems ?? this.extractedItems,
      confidenceScore: confidenceScore ?? this.confidenceScore,
      verified: verified ?? this.verified,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: apps/api/src/domain/models/receipt.rs
use chrono::{NaiveDate, NaiveDateTime};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Copy, Serialize, Deserialize, sqlx::Type)]
#[sqlx(type_name = "enum", rename_all = "UPPERCASE")]
pub enum OcrStatus {
    Pending,
    Processing,
    Completed,
    Failed,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Receipt {
    pub id: i64,
    pub transaction_id: Option<i64>,
    pub user_id: i64,
    pub group_id: Option<i64>,
    pub image_url: String,
    pub ocr_result: Option<serde_json::Value>,
    #[sqlx(rename = "ocr_status")]
    pub ocr_status: OcrStatus,
    pub extracted_amount: Option<i64>,
    pub extracted_date: Option<NaiveDate>,
    pub extracted_merchant: Option<String>,
    pub extracted_items: Option<serde_json::Value>,
    pub confidence_score: Option<rust_decimal::Decimal>,
    pub verified: bool,
    #[sqlx(default)]
    pub created_at: NaiveDateTime,
    #[sqlx(default)]
    pub updated_at: NaiveDateTime,
}

